Feature/ticket 12/borrowing system refactor

salet-gutierrez-jalafund opened the pull request:
Taiga

[https://tree.taiga.io/project/santiago_c_saavedra-fakeorgasm/us/12](https://tree.taiga.io/project/santiago_c_saavedra-fakeorgasm/us/12)

Commit:
refactor: split ReturnBook method to meet (SRP)

Commit:
refactor: split LendBook method to meet (SRP)

Commit:
Merge branch 'develop' into feature/ticket-12/borrowing-system-refactor

Commit:
refactor: add private access modifier to Loan class attributes

Change Book and Patron to class ids

Commit:
fix: change Book and Patron ids to match Loan class

Commit:
fix: change login to Patron id

Commit:
fix: change GetLoansByPatron and GetActiveLoansByPatron methods to Patron id

Commit:
fix: change to the ids of the Loan class; add search for a Patron or Book according to their ids

Commit:
fix: add searches according to Book or Pattern ids; add repository attributes to perform the searches

Commit:
feat: add console print format

Add book repository and repository pattern attributes to search by id and print the corresponding title and name.

Commit:
fix: add book and pattern repositories in the Reposter class

Adding the repositories will help to search for the ids and then return the correct response according to the id found, to provide a report through the console according to what the user chose.

Commit:
fix: fix errors in the Program class

Add the necessary parameters so that the Program class can compile and output the result satisfactorily.

Commit:
feat: add NoWarn in .csproj file

Commit:
refactor: remove warning in LoanControllerAsText class

Commit:
refactor: add access modifiers to the Fine class

Commit:
feat: add LoanFormater class to your console format

Commit:
refactor: add LoanFormatter class; add FormatLoan method to print to console

Commit:
feat: add the LoanFormatter class in the Console Line Renderer class

Add the FormatLoan method in the methods required to print to the console according to the Loan format and with its Fine attributes

Commit:
refactor: add ILoanRepository attribute in GetFinesByPatron method in IFineRepository interface

Commit:
feat: add functionality according to the idLoan, performing your search with that id

Commit:
refactor: add loan id

Commit:
refactor: add ILoanRepository parameter

Commit:
refactor: add the HasFine method to the loan Repository parameter

Commit:
refactor: add search by Loan id

Commit:
refactor: add parameters to classes that are in Program

Commit:
Merge branch 'feature/ticket-08/database-implementation-dapper' into feature/ticket-12/borrowing-system-refactor

Commit:
refactor: add functionality to database methods

Functionalities are being added to the database and the tasks that the database is performing have also been brought in

Commit:
refactor: split the functionality of the LendBook and ReturnBook method

Commit:
refactor: remove Loan console format

Commit:
feat: add data to the database

axel-ayala-jalafund commented:
@salet-gutierrez-jalafund  About coding I don't have observations, but about `init.sql` file that start the data from the `docker-compose.yml` is not working when the database was initialized previously. 
So, you need to update the README file instructions for this cases.

Basically, if is the first time that is initializing the database, the data will charge normally. But if already exists, you need to access into de container, that's are the steps:

_This steps are only if the data into de program or system are not showing_
Access to container:
```
 docker exec -it fakeorgasm-db bash
```
Connect to database:
```
psql -U fakeLibrary -d FakeOrgasm
```
And execute the script manually:
```
\i /docker-entrypoint-initdb.d/init.sql
```

Commit:
feat: add database instructions

The instructions explain if a container has been created what the user should do so that they can see the corresponding data.
